projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/CustodyDateUpdateService.kt

```diff
@@ -33,6 +34,10 @@ class CustodyDateUpdateService(
         }
     }
 
+    fun findNomsByCrn(crn: String): String {
+        return personRepository.getPersonWithNomsId(crn).nomsId!!
```

Comment:
This will fail if a CRN doesn't have a corresponding NOMS ID - which will happen a lot.  In that case we should ignore the message, as there'll be no key dates to update.

Also might be worth making this a JPA query repository method (e.g. `"select p.nomsId from Person p where crn = :crn"`), so you're not pulling back the whole person entity.

Reply:
Im throwing a not found exception in the getPersonWithNomsId when a record isnt found. I thought that this would be better than handling in the service

Reply:
changed